src/java.base/share/classes/java/lang/constant/ClassDesc.java

```diff
@@ -78,6 +78,27 @@ static ClassDesc of(String name) {
         return ClassDesc.ofDescriptor("L" + binaryToInternal(name) + ";");
     }
 
+    /**
+     * Returns a {@linkplain ClassDesc} for a class or interface type,
+     * given the name of the class or interface in internal form,
+     * such as {@code "java/lang/String"}.
+     * (To create a descriptor for an array type, either use {@link #ofDescriptor(String)}
```

Comment:
It feels like this section could be expanded a bit (and parenthesis removed). E.g. as you point out in the comment, this method only supports reference types, so we should state so clearly. And then present alternatives (like you do now).

Reply:
Isn’t the internal name of an array type its descriptor though?

So, I feel like this method should allow array descriptors.

Reply:
First may I ask which parenthesis should be removed?

Mentioning "reference" is misleading when talking about Class internal name (JVMS 4.2.1 "Binary Class and Interface Names"). Word "reference" is mentioned in (JVMS 4.3.2. "Field Descriptors", Table 4.3-A. "Interpretation of field descriptors") and it also includes array types descriptors, so it might be rather mentioned in ClassDesc::ofDescriptor method doc.

This is also answer to the second comment - field descriptor of array type isn't an internal name according to my understanding of JVMS.
This method suppose to construct ClassDesc from class or interface internal name according to JVMS 4.2.1.
Construction of ClassDesc from field descriptor according to JVMS 4.3.2 is ClassDesc::ofDescriptor method responsibility.